scrapy/utils/iterators.py

```diff
@@ -125,7 +125,7 @@ def _getrow(csv_r):
 
 
 def _body_or_str(obj, unicode=True):
-    assert isinstance(obj, (Response, six.string_types)), \
+    assert isinstance(obj, (Response, six.string_types, bytes)), \
```

Comment:
Nitpicking: if this function really should accept both text and binary, I'd make it explicit with `six.text_type, six.binary_type` instead of `six.string_types, bytes`, and use those to build the assertion message too (since `basestring` doesn't exist in Python 3).